gax-java/gax/src/test/java/com/google/api/gax/retrying/ScheduledRetryingExecutorTest.java

```diff
@@ -185,75 +174,64 @@ void testSuccessWithFailuresGetAttempt() throws Exception {
       assertFutureSuccess(future);
       assertEquals(15, future.getAttemptSettings().getAttemptCount());
       assertTrue(checks > 1 && checks <= maxRetries, "checks is equal to " + checks);
-      localExecutor.shutdownNow();
     }
   }
 
-  @ParameterizedTest
-  @MethodSource("data")
-  void testCancelGetAttempt(boolean withCustomRetrySettings) throws Exception {
-    setUp(withCustomRetrySettings);
-    for (int executionsCount = 0; executionsCount < EXECUTIONS_COUNT; executionsCount++) {
-      ScheduledExecutorService localExecutor = Executors.newSingleThreadScheduledExecutor();
-      final int maxRetries = 20;
-
-      FailingCallable callable = new FailingCallable(maxRetries - 1, "request", "SUCCESS", tracer);
-      RetrySettings retrySettings =
-          FAST_RETRY_SETTINGS
-              .toBuilder()
-              .setTotalTimeoutDuration(java.time.Duration.ofMillis(5000L))
-              .setMaxAttempts(maxRetries)
-              .build();
-
-      RetryingExecutorWithContext<String> executor =
-          getRetryingExecutor(getAlgorithm(retrySettings, 0, null), localExecutor);
-      RetryingFuture<String> future = executor.createFuture(callable, retryingContext);
-      callable.setExternalFuture(future);
-
-      assertNull(future.peekAttemptResult());
-      assertSame(future.getAttemptResult(), future.getAttemptResult());
-      assertFalse(future.getAttemptResult().isDone());
-      assertFalse(future.getAttemptResult().isCancelled());
-
-      future.setAttemptFuture(executor.submit(future));
-
-      CustomException exception;
-      CancellationException cancellationException = null;
-      int checks = 0;
-      int failedCancelations = 0;
-      do {
-        exception = null;
-        checks++;
-        Future<String> attemptResult = future.getAttemptResult();
-        try {
-          attemptResult.get();
-          assertNotNull(future.peekAttemptResult());
-        } catch (CancellationException e) {
-          cancellationException = e;
-        } catch (ExecutionException e) {
-          exception = (CustomException) e.getCause();
-        }
-        assertTrue(attemptResult.isDone());
-        if (!future.cancel(true)) {
-          failedCancelations++;
-        }
-      } while (exception != null && checks < maxRetries);
-
-      assertTrue(future.isDone());
-      assertNotNull(cancellationException);
-      // future.cancel(true) may return false sometimes, which is ok. Also, the every cancellation
-      // of
-      // an already cancelled future should return false (this is what -1 means here)
-      assertEquals(2, checks - (failedCancelations - 1));
```

Comment:
Is this the failing point for #2851? Do you think it is redundant?

Reply:
I'm not sure what you're referring to. The test is to ensure that the future was cancelled during the retry attempt.

Future should be reported as done and that one attempt was made.